### DISCUSSION TITLE
Player crops portrait videos on default configs

### DISCUSSION THREAD

**Discussion by victorges** [livepeer/livepeer.js#87] [Ideas]
Our default configuration options on the player currently use:
 - Aspect ratio: 16:9 (this enforces the aspect ratio of the player, even if the video is in a different ratio)
 - Object fit: `cover` (which means "fill" instead of "fit" to area)

While this works well for landscape videos. If they are not exactly on 16:9 aspect ratio, they will be cropped on the sides to fill the player area, which can look better than adding black bars to them.

With portrait videos tho, this effectively means that we crop around ~70% of the video, displaying only the very center of it. This means that applications that want to show portrait videos will need to make custom configurations on the player to get minimum functionality working.

IMO the best fix for this would be defaulting the Object Fit to `contain`, which means to fit the video on the player area adding black bars to the sides. This will work well on all cases instead, by also adding black bars to landscape videos not exactly 16:9, which is a fine compromise to me. This is not the consensus tho, so let's talk about it to decide.

**Comment by 0xcadams** [comment]
See here for details. Moving this to a discussion. https://livepeerjs.org/react/Player#objectfit

**Comment by 0xcadams** [comment]
@adamsoffer tagging you for visibility

**Comment by victorges** [comment]
Cross-linking a comment about my view on this matter: https://github.com/livepeer/studio/pull/1330#issuecomment-1270612102

TLDR (IMO) players should (default to) displaying the entire content of a video, in the best way possible. Cropping a video is not doing that, as it focus more on making the player look good, with no black bars, rather than displaying the source content to the user.

**Reply by 0xcadams** [reply]
Yeah, the studio player is another example of the content not being curated by the developer who is creating the page. So, it totally makes sense to add `contain` there. Not sure that our internal use-cases are the best example since we don't know the source content aspect ratio. I don't mean to be annoying with this, I would just like to hold off until we start getting some real use for this SDK - I don't see why this is so important since we provide the `objectFit`, and browsers by default use `fill`.

**Reply by victorges** [reply]
I believe this is gonna be a point of friction for any developer that allows videos with different aspect ratios in their applications. It also behaves in a much more silent way, which devs might not even notice that it is happening until some user reports that their uploaded videos are broken/cropped and only then they find out it is a player config, not broken video.

So I think it makes more sense to default to a behavior that can be more easily spotted and doesn't look like a bug (e.g. "Livepeer processed my video wrong"). I don't think we should expect developers to read the entirety of the documentation before getting started. I bet they'll end up copying the first example and iterating on the code suggestions on the IDE.

Just to explain where I'm coming from as well, I was really baffled by the experience of uploading a portrait video on the dashboard then seeing it completely cropped on the player. I don't think that is a reasonable default. It just feels broken, which is not the same experience of a black bar IMO.

**Comment by 0xcadams** [comment]
Related: https://github.com/livepeer/livepeer.js/pull/197 😄

**Comment by victorges** [comment]
Nice! So I guess this one can be closed now?
